src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_broodlord_lashlayer.cpp

```diff
@@ -113,7 +122,86 @@ class boss_broodlord : public CreatureScript
     }
 };
 
+class go_suppression_device : public GameObjectScript
+{
+    public:
+        go_suppression_device() : GameObjectScript("go_suppression_device") { }
+
+        struct go_suppression_deviceAI : public GameObjectAI
+        {
+            go_suppression_deviceAI(GameObject* go) : GameObjectAI(go), _instance(go->GetInstanceScript()) 
+            {
+                _events.ScheduleEvent(EVENT_SUPPRESSION_AURA, Seconds(6));
+                _events.ScheduleEvent(EVENT_SUPPRESSION_ANIM, Seconds(2));
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                if (_instance->GetBossState(DATA_VAELASTRAZ_THE_CORRUPT) != DONE)
+                    return;
+
+                if (_instance->GetBossState(DATA_VAELASTRAZ_THE_CORRUPT) == DONE)
+                    {
+                        if (me->GetGoState() == GO_STATE_DESTROYED)
+                        me->SetLootState(GO_READY); // NOT doing this will despawn the gameobject after using Disarm Trap
+                    }
+
+                if (_instance->GetBossState(DATA_BROODLORD_LASHLAYER) == DONE)
+                    {
```

Comment:
formatting

Reply:
these checks running each update diff does not seem the most appropriate

Reply:
Im up for suggestions here but i couldnt come up with something else.
The intention was:
Do nothing until VAEL is dead.
Execute some actions and then do nothing after BROOD is dead.

The check inbetween these 2 is needed, if we dont do this, the gameobject is despawned by 'Disarm Trap'.